sycl/source/detail/queue_impl.cpp

```diff
@@ -35,33 +37,48 @@ template <> device queue_impl::get_info<info::queue::device>() const {
 
 event queue_impl::memset(void *Ptr, int Value, size_t Count) {
   context Context = get_context();
-  std::shared_ptr<usm::USMDispatcher> USMDispatch =
-      getSyclObjImpl(Context)->getUSMDispatch();
-  cl_event Event;
+  if (Context.is_host()) {
+    std::memset(Ptr, Value, Count);
 
-  PI_CHECK(USMDispatch->enqueueMemset(getHandleRef(), Ptr, Value, Count,
-                                      /* sizeof waitlist */ 0, nullptr,
-                                      reinterpret_cast<pi_event *>(&Event)));
+    return event();
+  } else {
```

Comment:
This else is unnecessary.
https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return